### PR TITLE
Add Timed helm chart

### PR DESCRIPTION
**Pull request by sliverc**
An initial helm chart running Timed backend with an external database. This is a functional helm chart but not a complete Timed chart yet. Current TODOs are document in README.md.

### CODE REVIEW

File: Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+description: Chart for Timed application
+name: timed
+version: 0.1.0
+keywords:
+  - timed
+  - tracking
+sources:
+  - https://github.com/adfinis-sygroup/timed-chart
```
Comment (tongpu) [inline Chart.yaml:9]: I would also add a link to the timed source code here.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,37 @@
...
+To uninstall/delete the `my-release` deployment:
+
+```bash
+$ helm delete my-release
+```
+
+## TODOs
+
+* Integrate Timed Frontend
+* Deploy postgres as well
```
Comment (tongpu) [inline README.md:34]: This could probably be solved by adding the postgresql chart as a dependency.

File: templates/pre-install-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "name" . }}
+  # This declares the resource to be a hook. By convention, we also name the
+  # file "pre-install-XXX.yaml", but Helm itself doesn't care about file names.
+  annotations:
+    "helm.sh/hook": pre-install
+type: Opaque
+data:
+  {{ if .Values.service.mysqlRootPassword }}
```
Comment (tongpu) [inline templates/pre-install-secret.yaml]: I don't think `mysqlRootPassword` belongs here.

**Review by tongpu — CHANGES_REQUESTED** [review]
Just a quick review of the files included in the chart. I didn't yet have time to run the chart.

File: Chart.yaml (hunk 1 above)
Comment (sliverc) [reply Chart.yaml:9]: Make sense. Added.

File: README.md (hunk 2 above)
Comment (sliverc) [reply README.md:34]: I will try this in a later PR.

File: templates/pre-install-secret.yaml (hunk 3 above)
Comment (sliverc) [reply templates/pre-install-secret.yaml]: Certainly not. Changed.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+*.swp
```
Comment (hairmare) [inline .gitignore]: You might want to consider using a [global .gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) for $EDITOR specific files.

File: .helmignore [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# Common VCS dirs
+.git/
+.gitignore
+.bzr/
+.bzrignore
+.hg/
+.hgignore
+.svn/
+# Common backup files
+*.swp
```
Comment (hairmare) [inline .helmignore]: `/*.swp/*.sw?/`

This is probably from a generated file as well. I know that I should take more care not to let my $EDITOR create multiple swap files, making this glob bit more generic would help anyway.

File: .gitignore (hunk 4 above)
Comment (sliverc) [reply .gitignore]: This is a default file helm created for me... :) but fair point I think we would need to discuss it for all projects.

**Comment by sliverc** [comment]
@hairmare I have fixed the helm linting errors.

**Review by hairmare — COMMENTED** [review]
LGTM... It's not like I understand most of it, much  helm learning will still be needed so I'll have to defer to @tongpu for any serious review stuff.

File: .helmignore (hunk 5 above)
Comment (sliverc) [reply .helmignore]: Changed this.

**Review by tongpu — APPROVED** [review]

**Comment by tongpu** [comment]
LGTM too. Further refinement will happen in future PRs.

**Pull request merged**